src/testdir/test_let.vim

```diff
@@ -25,3 +25,55 @@ func Test_let()
   let s = "\na                     #1\nb                     #2"
   call assert_equal(s, out)
 endfunc
+
+func s:set_arg1(a) abort
+  let a:a = 1
+endfunction
+
+func s:set_arg2(a) abort
+  let a:b = 1
+endfunction
+
+func s:set_arg3(a) abort
+  let b = a:
+  let b['a'] = 1
+endfunction
+
+func s:set_arg4(a) abort
+  let b = a:
+  let b['a'] = 1
+endfunction
+
+func s:set_arg5(a) abort
+  let b = a:
+  let b['a'][0] = 1
+endfunction
+
+func s:set_arg6(a) abort
+  let a:a[0] = 1
+endfunction
+
+func s:set_arg7(a) abort
+  call extend(a:, {'a': 1})
+endfunction
+
+func s:set_arg8(a) abort
+  call extend(a:, {'b': 1})
+endfunction
+
+func s:set_arg9(a) abort
+  let a:['b'] = 1
+endfunction
+
```

Comment:
please add 

```vim
function s:set_arg10(...) abort
  let a:000[0][0] = 1
endfunction

call s:set_arg10([0])
```

Reply:
Thank you. I added `Test_let_varg_fail()` for that.